via_verifier/lib/via_musig2/src/utxo_manager.rs

```diff
@@ -0,0 +1,426 @@
+use std::{collections::VecDeque, sync::Arc};
+
+use bitcoin::{Address, Amount, OutPoint, Transaction, TxOut};
+use tokio::sync::RwLock;
+use via_btc_client::traits::BitcoinOps;
+
+const CTX_REQUIRED_CONFIRMATIONS: u32 = 1;
+const DEFAULT_CAPACITY: usize = 100;
+
+#[derive(Debug, Clone)]
+pub struct UtxoManager {
```

Comment:
Maybe in the future we can try to make this generic enough so it can handle the InsciberContext as well, but looks good for now.